celery/canvas.py

```diff
@@ -1298,7 +1298,14 @@ def run(self, header, body, partial_args, app=None, interval=None,
         else:
             body.delay([])
 
-        bodyres.parent = header_result
+        bodyres_root = bodyres
+        seen = set()
+        while bodyres_root.parent is not None:
```

Comment:
Hi @Prilkop .parent can be None in this line?

Reply:
can you tackle this one with appropriate test?